tidb: add a new command line flag for redact (#17262)
pingcap/docs: command-line-flags-for-tidb-configuration.md

+ Sets the CPU affinity of TiDB servers, which is separated by commas. For example, "1,2,3".
+ Default: `""`

## `--redact`

> + Determines whether TiDB server desensitizes log files when using the subcommand `collect-log`.

Reviewer: ```suggestion
+ Determines whether the TiDB server desensitizes log files when using the subcommand `collect-log`.
```

> + Determines whether to enable TiDB server to desensitize log files.
> + Default: false
> + You can use `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the `<input>` log file and output it to `<output>`. When the value is `true`, TiDB desensitizes the log file. When it is `false`, TiDB restores the log file. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).

Author: ```suggestion
+ Determines whether TiDB server desensitizes log files when using the subcommand `collect-log`.
+ Default: false
+ When the value is `true`, it is a masking operation, and all fields marked with `‹..›` are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `< output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).
```

> + When the value is `true`, it is a masking operation, and all fields wrapped in `‹ ›` are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `<output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).

Reviewer: ```suggestion
+ When the value is `true`, it is a masking operation, and all fields wrapped in `‹ ›` mark symbols are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `<output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).
```

> + When the value is `true`, it is a masking operation, and all fields marked with `‹..›` are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `<output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).

Author: ```suggestion
+ When the value is `true`, it is a masking operation, and all fields wrapped in `‹ ›` are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `<output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).
```

> + When the value is `true`, it is a masking operation, and all fields marked with `‹..›` are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `< output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).

Reviewer 2: ```suggestion
+ When the value is `true`, it is a masking operation, and all fields marked with `‹..›` are replaced with `?`. When the value is `false`, it is a restore operation, and all mark symbols are removed. To use this feature, execute `./tidb-server --redact=xxx collect-log <input> <output>` to desensitize or restore the TiDB server log file specified by `<input>` and output it to `<output>`. For more information, see the system variable [`tidb_redact_log`](/system-variables.md#tidb_redact_log).
```
